### ISSUE TITLE
Question about UniformLearner

### ISSUE THREAD

**Issue by lydiaji** [Tencent/PocketFlow#319]
Hello,  I'm curious about why here use "
   w_max = tf.stop_gradient(tf.reduce_max(w, axis=axis))
   w_min = tf.stop_gradient(tf.reduce_min(w, axis=axis))
"
when evaluating, the min, max value do not exist in model file,  why can we quantize like the below function? Thanks.

  def __scale(self, w, mode):
    """linear scale function

    Args:
    * w: A Tensor (weights or activation output),
         the shape is [bucket_size, bucekt_num] if use_buckets else the original size.
    * mode: A string, 'weight' or 'activation'

    Returns:
    * A Tensor, the normalized weights
    * A Tensor, alpha, scalar if activation mode else a vector [bucket_num].
    * A Tensor, beta, scalar if activation mode else a vector [bucket_num].
    """
    if mode == 'weight':
      if self.use_buckets:
        axis = 0
      else:
        axis = None
    elif mode == 'activation':
      axis = None
    else:
      raise ValueError("Unknown mode for scalling")

    w_max = tf.stop_gradient(tf.reduce_max(w, axis=axis))
    w_min = tf.stop_gradient(tf.reduce_min(w, axis=axis))
    eps = tf.constant(value=1e-10, dtype=tf.float32)

    alpha = w_max - w_min + eps
    beta = w_min
    w = (w - beta) / alpha
    return w, alpha, beta

**Comment by yuanyuanli85** [comment]
The implementation in this repo does not take this into consideration.  In real case, either keeping max/min as variable or using moving average (just like bn) can help here.

**Comment by lydiaji** [comment]
> in

I see. Thanks so much ^^
